test_backend_batch_statistics_reset failure in assert test_time / ...

joblib package keeps failing in debian reproducible builds with the following test failure:
https://tests.reproducible-builds.org/debian/rbuild/unstable/i386/joblib_0.12.2-1.rbuild.log.gz
([package page](https://tests.reproducible-builds.org/debian/rb-pkg/unstable/i386/joblib.html))
```
        # Tolerance in the timing comparison to avoid random failures on CIs
>       assert test_time / ref_time <= 1 + relative_tolerance
E       assert (2.2405660152435303 / 1.3164479732513428) <= (1 + 0.2)
```
Any hints on what potentially could have lead to it and how to mitigate are more than welcome!

For the record, this is the full traceback of the failure

```
=================================== FAILURES ===================================
_____________ test_backend_batch_statistics_reset[multiprocessing] _____________

backend = 'multiprocessing'

    @with_multiprocessing
    @parametrize('backend', ['multiprocessing', 'loky'])
    def test_backend_batch_statistics_reset(backend):
        """Test that a parallel backend correctly resets its batch statistics."""
        relative_tolerance = 0.2
        n_jobs = 2
        n_inputs = 500
        task_time = 2. / n_inputs
    
        p = Parallel(verbose=10, n_jobs=n_jobs, backend=backend)
        start_time = time.time()
        p(delayed(time.sleep)(task_time) for i in range(n_inputs))
        ref_time = time.time() - start_time
        assert (p._backend._effective_batch_size ==
                p._backend._DEFAULT_EFFECTIVE_BATCH_SIZE)
        assert (p._backend._smoothed_batch_duration ==
                p._backend._DEFAULT_SMOOTHED_BATCH_DURATION)
    
        start_time = time.time()
        p(delayed(time.sleep)(task_time) for i in range(n_inputs))
        test_time = time.time() - start_time
        assert (p._backend._effective_batch_size ==
                p._backend._DEFAULT_EFFECTIVE_BATCH_SIZE)
        assert (p._backend._smoothed_batch_duration ==
                p._backend._DEFAULT_SMOOTHED_BATCH_DURATION)
    
        # Tolerance in the timing comparison to avoid random failures on CIs
>       assert test_time / ref_time <= 1 + relative_tolerance
E       assert (2.2405660152435303 / 1.3164479732513428) <= (1 + 0.2)

joblib/test/test_parallel.py:1206: AssertionError
============= 1 failed, 1070 passed, 13 skipped in 133.04 seconds ==============
```

Interesting, [on amd64](https://tests.reproducible-builds.org/debian/rb-pkg/buster/amd64/joblib.html), the error is with the 'loky' backend:
```
=================================== FAILURES ===================================
__________________ test_backend_batch_statistics_reset[loky] ___________________

backend = 'loky'

    @with_multiprocessing
    @parametrize('backend', ['multiprocessing', 'loky'])
    def test_backend_batch_statistics_reset(backend):
        """Test that a parallel backend correctly resets its batch statistics."""
        relative_tolerance = 0.2
        n_jobs = 2
        n_inputs = 500
        task_time = 2. / n_inputs
    
        p = Parallel(verbose=10, n_jobs=n_jobs, backend=backend)
        start_time = time.time()
        p(delayed(time.sleep)(task_time) for i in range(n_inputs))
        ref_time = time.time() - start_time
        assert (p._backend._effective_batch_size ==
                p._backend._DEFAULT_EFFECTIVE_BATCH_SIZE)
        assert (p._backend._smoothed_batch_duration ==
                p._backend._DEFAULT_SMOOTHED_BATCH_DURATION)
    
        start_time = time.time()
        p(delayed(time.sleep)(task_time) for i in range(n_inputs))
        test_time = time.time() - start_time
        assert (p._backend._effective_batch_size ==
                p._backend._DEFAULT_EFFECTIVE_BATCH_SIZE)
        assert (p._backend._smoothed_batch_duration ==
                p._backend._DEFAULT_SMOOTHED_BATCH_DURATION)
    
        # Tolerance in the timing comparison to avoid random failures on CIs
>       assert test_time / ref_time <= 1 + relative_tolerance
E       assert (1.4412801265716553 / 1.0927374362945557) <= (1 + 0.2)

joblib/test/test_parallel.py:1150: AssertionError
============== 1 failed, 1095 passed, 8 skipped in 126.84 seconds ==============
```

--- a/joblib/test/test_parallel.py
+++ b/joblib/test/test_parallel.py
@@ -1191,30 +1191,22 @@ def test_delayed_check_pickle_deprecated():
 @parametrize('backend', ['multiprocessing', 'loky'])
 def test_backend_batch_statistics_reset(backend):
     """Test that a parallel backend correctly resets its batch statistics."""
-    relative_tolerance = 0.2
     n_jobs = 2
     n_inputs = 500
     task_time = 2. / n_inputs
 
     p = Parallel(verbose=10, n_jobs=n_jobs, backend=backend)
-    start_time = time.time()
     p(delayed(time.sleep)(task_time) for i in range(n_inputs))
-    ref_time = time.time() - start_time
     assert (p._backend._effective_batch_size ==
             p._backend._DEFAULT_EFFECTIVE_BATCH_SIZE)
     assert (p._backend._smoothed_batch_duration ==
             p._backend._DEFAULT_SMOOTHED_BATCH_DURATION)
 
-    start_time = time.time()
     p(delayed(time.sleep)(task_time) for i in range(n_inputs))
-    test_time = time.time() - start_time
     assert (p._backend._effective_batch_size ==
             p._backend._DEFAULT_EFFECTIVE_BATCH_SIZE)
     assert (p._backend._smoothed_batch_duration ==
             p._backend._DEFAULT_SMOOTHED_BATCH_DURATION)
-
-    # Tolerance in the timing comparison to avoid random failures on CIs
-    assert test_time / ref_time <= 1 + relative_tolerance
 
 
 def test_backend_hinting_and_constraints():

--- a/joblib/test/test_parallel.py
+++ b/joblib/test/test_parallel.py
@@ -1383,7 +1383,7 @@ def test_nested_parallelism_limit(backend):
 @with_numpy
 def test_nested_parallelism_with_dask():
     distributed = pytest.importorskip('distributed')
-    client = distributed.Client()  # noqa
+    client = distributed.Client(n_workers=2, threads_per_worker=2)  # noqa
 
     # 10 MB of data as argument to trigger implicit scattering
     data = np.ones(int(1e7), dtype=np.uint8)

